### PR TITLE
Required helper methods to support chef-ingredient

### PR DESCRIPTION
**Pull request by sersut**
This PR adds the required helper methods to use mixlib-install in chef-ingredient.

/cc: @schisamo, @patrick-wright, @thommay, @jtimberman 

Note that there are more things to move from mixlib-install to here but they will be moved in a different PR.

### CODE REVIEW

**Comment by patrick-wright** [comment]
:+1:

**Comment by sersut** [comment]
Travis failures are only on `ruby-head` and not related:

```
Libraries missing for ruby-head: libtcl8.5.so.0,libtk8.5.so.0. Refer to your system manual for installing libraries
Mounting remote ruby failed with status 10, stopping installation.
The command "rvm install ruby-head --binary" failed and exited with 1 during .
```

File: lib/mixlib/install.rb [hunk 1]
```diff
@@ -59,5 +61,32 @@ def root
...
+      if File.exist? version_manifest_file
+        JSON.parse(File.read(version_manifest_file))["build_version"]
+      end
+    end
+
+    #
+    # Returns true if an upgradable version is available, false otherwise.
+    #
+    def upgrade_available?
+      return false if current_version.nil?
```
Comment (schisamo) [inline lib/mixlib/install.rb]: Wouldn't we just assume an upgrade is available if the current version can't be determined?

Comment (sersut) [inline lib/mixlib/install.rb]: Ahh... Nice catch @schisamo. :+1:

**Comment by curiositycasualty** [comment]
:+1:

File: lib/mixlib/install.rb [hunk 2]
```diff
@@ -59,5 +61,32 @@ def root
...
+    #
+    # Returns the current version of the installed product.
+    # Returns nil if the product is not installed.
+    #
+    def current_version
+      # Note that this logic does not work for products other than
+      # chef & chefdk since version-manifest is created under the
+      # install directory which can be different than the product name (e.g.
+      # chef-server -> /opt/opscode). But this is OK for now since
+      # chef & chefdk are the only supported products for now.
```
Comment (jtimberman) [inline lib/mixlib/install.rb]: too many "for now"s in this comment :smile:

Comment (sersut) [inline lib/mixlib/install.rb]: :) :+1:

**Pull request merged**